kasa/discover.py

```diff
@@ -37,86 +108,76 @@ class Discover:
     }
 
     @staticmethod
-    def discover(
-        protocol: TPLinkSmartHomeProtocol = None,
-        target: str = "255.255.255.255",
-        port: int = 9999,
-        timeout: int = 3,
+    async def discover(
+        *,
+        target="255.255.255.255",
+        on_discovered=None,
```

Comment:
A type hint might be useful here. 

OTOH it's a private class.

Reply:
Indeed; this wasn't completely done. I'll make the changes in the other PR considering it already contains all of this. So I'm closing this now, thanks for the review!